### DISCUSSION TITLE
Changing NFT Supply After Deployment.

### DISCUSSION THREAD

**Discussion by tomvanriet** [TENK-DAO/tenk#26] [Q&A] [answered]
Sorry if this is a silly question, but how would one change the total supply after the contract has been deployed?

**Comment by willemneal** [comment, marked as answer]
Currently the raffle data structure is set to an initial range.  So there are three potential situations:

1. **All the tokens have been minted.**  In this case you just need to create a new raffle with a range that starts at the end of the first. Then minting can resume as normal.

2. **Not all tokens have been minted.**  There are two options: a) don't do anything until all tokens are minted and then return to 1. b) This one is currently tricky.  You would need to update the raffle option to allow for two sets of possible ranges.  It's possible, but takes some thought to do it correctly given the new probabilities.

3. **No tokens have been minted.** This is the easiest route and one that I assume the question is not meant for, but if you have it deployed and no mints have taken place, then you just need to update the range of the raffle.



In all of these scenarios you also need to update the new `base_uri` that points to a directory with the new total supply of assets.

**Reply by tomvanriet** [reply]
Thanks so much Willem!
